src/Nethermind/Nethermind.Network/P2P/Subprotocols/Eth/V62/TxFloodController.cs

```diff
@@ -1,72 +1,93 @@
 // SPDX-FileCopyrightText: 2025 Demerzel Solutions Limited
 // SPDX-License-Identifier: LGPL-3.0-only
 
+using System;
 using Nethermind.Core;
 using Nethermind.Logging;
 using Nethermind.Stats.Model;
 using Nethermind.TxPool;
-using System;
 
-namespace Nethermind.Network.P2P.Subprotocols.Eth.V62
+namespace Nethermind.Network.P2P.Subprotocols.Eth.V62;
+
+internal class TxFloodController(Eth62ProtocolHandler protocolHandler, ITimestamper timestamper, ILogger logger)
 {
-    internal class TxFloodController(Eth62ProtocolHandler protocolHandler, ITimestamper timestamper, ILogger logger)
-    {
-        private DateTime _checkpoint = DateTime.UtcNow;
-        private readonly Eth62ProtocolHandler _protocolHandler = protocolHandler ?? throw new ArgumentNullException(nameof(protocolHandler));
-        private readonly ITimestamper _timestamper = timestamper ?? throw new ArgumentNullException(nameof(timestamper));
-        private readonly TimeSpan _checkInterval = TimeSpan.FromSeconds(60);
-        private long _notAcceptedSinceLastCheck;
-        private readonly Random _random = new();
+    // Late with one block
+    private const int DegradeRatePerMinute = 250;
+    // Late with 5 blocks (and has all included txs in their pool)
+    private const int DisconnectRatePerMinute = 750;
 
-        internal bool IsDowngraded { get; private set; }
+    private DateTime _checkpoint = DateTime.UtcNow;
+    private readonly Eth62ProtocolHandler _protocolHandler = protocolHandler ?? throw new ArgumentNullException(nameof(protocolHandler));
+    private readonly ITimestamper _timestamper = timestamper ?? throw new ArgumentNullException(nameof(timestamper));
+    private readonly TimeSpan _checkInterval = TimeSpan.FromSeconds(60);
+    private long _notAcceptedSinceLastCheck;
+    private readonly Random _random = new();
 
-        public void Report(AcceptTxResult accepted)
-        {
-            TryReset();
+    internal bool IsDowngraded { get; private set; }
 
-            if (!accepted)
+    public void Report(in AcceptTxResult accepted)
```

Comment:
Can we rename? `accepted` suggest it was accepted, while we are explicitly checking for rejections

```suggestion
    public void Report(in AcceptTxResult acceptedResult)
```